ntbs-service/wwwroot/source/Components/ValidateMultiple.ts

```diff
@@ -66,6 +84,23 @@ function buildKeyValuePairsQueryString(keys: Array<string>, values: Array<string
     return qs.stringify(queryStringObject);
 };
 
+function buildKeyDateValuesQueryString(keys: Array<string>, days: Array<string>, months: Array<string>, years: Array<string>) {
+    const queryStringObject: any = { keyValuePairs: [] };

```

Comment:
another any :(

Reply:
But to fix this wouldn't I have to define a class for this particular object? Which seems a bit like overkill...

Reply:
no, you can just describe what type you'd like it to be, so somethign along the lines of

Reply:
No, you can just describe the type you want it to be, so something along the lines of
```
queryStringObject: {keyValuePairs: Array<{ key: string, day: string, month: string, year: string }>} = { keyValuePairs: [] }
```
or, if you want it to be more generic (and therefore the same for the method above, you can do 
```
queryStringObject: {keyValuePairs: Array<{[key: string]: string}>} = { keyValuePairs: [] }
```
... but, as I wrote that, it actually suggested to me that we've got a bit of an unnecessary wrapper going on - we could build up the array first and only place it in the object once we're done, like so (I've also pulled the type out above so it can be reused and would be easier to read):
```
type KeyValuePairs = Array<{[key: string]: string}>

function buildKeyDateValuesQueryString(keys: Array<string>, days: Array<string>, months: Array<string>, years: Array<string>): string {
    const keyValuePairs: KeyValuePairs = []
    for (let i = 0; i < keys.length; i++) {
        keyValuePairs.push({ key: keys[i], day: days[i], month: months[i], year: years[i] });
    }
    return qs.stringify({keyValuePairs});
};
```

Reply:
Great, that's much nicer!! Learnt a lot about typescript today, thanks  :)

Reply:
Yay! :) 🎉